server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -232,14 +271,28 @@ public String toString() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        return toXContent(builder, params, null);
+        return toXContent(builder, params, null, null);
     }
 
     /**
      * Converts the data stream lifecycle to XContent and injects the RolloverConditions if they exist.
+     * @deprecated use {@link #toXContent(XContentBuilder, Params, RolloverConfiguration, DataStreamGlobalRetention)}
      */
+    @Deprecated
```

Comment:
This is only used in 4 places -- might as well just remove this method rather than adding deprecation, right?

Reply:
It will be easier to track down the places that it is being used to properly update them in the follow up PR, because I will be able to use the `Find usages`.

It's a convenience thing and they will be removed in the next PR.

Of course, if you feel strongly about it I can just inline it.

Reply:
I'm good with removing them in a follow-up PR!